Friendly names & view all bugs

Aryllen opened the pull request:
Alright, after much testing and some bug fixes, this appears to be working.

In testing friendly names, I found some serious issues with the view all panel in the case that there were no submissions found for a given status. I updated synapseforms to send back NULL and updated stopadforms to handle these cases. These are, unfortunately, on the same branch due to the interesting fixes with merge conflicts and gitKraken being less flexible than I'd like when it comes to moving changes between branches.

Also updated the documentation.

karawoo commented on R/mod-panel-section.R:
```diff
@@ -74,29 +74,32 @@ mod_panel_section_ui <- function(id) {
   })
+  submission_name <- reactive({
+    reviews$submission[reviews$form_data_id == input$submission][1]
+  })
```
Does this work even though `reviews` isn't defined until the next line?

Aryllen commented:
Hmm, I don't seem to get those errors. I'll check into it. It also looks like the lines got messed up on order again. Going to have to stop using gitKraken for the conflicts because it's too difficult to tell if the lines are going to get reversed (or were reversed), particularly in large conflicts like this had.

I'll go back through all the code again and see where things are messed up.

Aryllen commented:
My local repo ended up all messed up. I cloned the repo from github, again, and fixed the problems mentioned, and a few more that I found. Tested on the shiny server (stopadforms-test) and all elements appear to be working. Sorry about the mess earlier.

A couple things:
- I cleaned out the section-comments table to not have all the stuff before friendly-names. This makes it so that the panel tab shows the friendly names and has less testing comments overall. We should clean this out again before giving it off to the clients.
- I cleaned out the submission-comments table for the same reasons as the section-comments table. Again, it should be cleaned out again before giving it off.

Please let me know if there's anything else that I didn't find this time around.
